src/screens/GroupInfo/hooks/useGroupInfo.js

```diff
@@ -97,12 +100,49 @@ const useGroupInfo = (channelId = null) => {
     return getChatName(username, profile.myProfile.username);
   };
   const chatName = getChatName(username, profile.myProfile.username);
-  const handleOnNameChange = () => {
-    navigation.push('GroupSetting', {
-      username: chatName,
-      focusChatName: true,
-      refresh: getMembersList
-    });
+  const handleOpenNameChange = () => {
+    setIsOpenModalChangeName(true);
+  };
+  const handleSaveNameChange = async (name) => {
+    setIsOpenModalChangeName(false);
+
+    try {
+      setIsUpdatingName(true);
+      const responseChannelData = await SignedMessageRepo.changeSignedChannelDetail(
+        channelId,
+        name,
+        null
+      );
+
+      const {channelName} = getChannelListInfo(
+        responseChannelData?.channel,
+        signedProfileId,
+        anonProfileId
+      );
+      const channelList = await ChannelListSchema.getSchemaById(localDb, channelId);
+      channelList.name = channelName;
+      channelList.rawJson = responseChannelData;
+      await channelList.save(localDb);
+
+      setSelectedChannel(channelList);
+      setIsUpdatingName(false);
+
+      refresh('channelList');
+      refreshWithId('chat', channelId);
+      refresh('channelInfo');
+
+      setTimeout(() => {
+        navigation.navigate('SignedChatScreen');
+      }, 500);
+    } catch (e) {
+      setIsUpdatingName(false);
+      if (__DEV__) {
+        console.log(e);
+      }
+    }
+  };
+  const closeOnNameChange = () => {
+    setIsOpenModalChangeName(false);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

A new function `handleSaveNameChange` has been introduced to handle the name change of a group. It updates the local database and refreshes the UI accordingly. Error handling is done properly with a try-catch block. However, it's recommended to provide user feedback in case of an error instead of just logging it.
```diff
+ const handleOpenNameChange = () => {
+   setIsOpenModalChangeName(true);
+ };
+ const handleSaveNameChange = async (name) => {
+   setIsOpenModalChangeName(false);
+   // ... rest of the code
+ };
+ const closeOnNameChange = () => {
+   setIsOpenModalChangeName(false);
+ };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->